### ISSUE TITLE
SimSwap must follow resource centric design described by camara APi guideline

### ISSUE THREAD

**Issue by gregory1g** [camaraproject/SimSwap#32]
Current API exposes a procedure - the API entry is a verb ("retrieve-date" or "check") with input and output parameters.
This contradicts chapter 4.1 of "API-design-guidelines" which describes APIs in a RESTFull style - where URIs identify resources and operations are exposes via HTTP verbs resources centric.

see: https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/API-design-guidelines.md

Suggestion: adjust the APIs to follow camara API design guidelines

**Comment by ShutingQing** [comment]
API guidelines needs to follow. But not to discuss about guidelines, just from a user's perspective, I feel like verb-noun can to large extent express the intention of one API. Verb only sometimes cause some confusions.

**Comment by gregory1g** [comment]
if we intentionally want to deviate from camara guidelines and RESTFul API style for this API - it make sense to document this decision and reasons behind.

Most probably, one can clearly express SimSwap API functionality using resource language as well.
Just as few example first came to my mind: 
sub-resource style: GET /sim-swap/v0/msisdns/+123345/recent_swap_date
query style: GET /sim-swap/v0/msisdns/+123345&mode=recent_swap_date

**Comment by monamok** [comment]
The situation here is that we agreed not to include user identifiers such as phone-numbers in the path, so we need to send them as request body. In that case, we can't use a GET and we would need a POST. If we use POST with the resource-name it would be confusing because that would mean that we're creating a resource which is not the case here. So that's why we need to use a verb to avoid the confussion. Using a POST when you are not creating wouldn't be REST-ful either.
You're right that if we used a GET we should have used nouns to make it resource-centric but in the current situation I think this is the best way to define it.

**Comment by gregory1g** [comment]
If there are good reasons to avoid MSISDNs be included in path and in the query, this pattern must be promoted to common design guidelines and followed by other APIs as well. 
Currently SimSwap API directly contradicts design guidelines, what is not great.

BTW, what are the reasons which motivated you to avoid using MSISDN in the path and query?
afaik, both are secured and hidden behind HTTPS same way as a request body. Therefore, from security point of view there is no difference,

just as an example, twilio SIM Swap API looks the following way: "GET https://lookups.dublin.ie1.twilio.com/v2/PhoneNumbers/+447772000001?Fields=sim_swap"

**Comment by ShutingQing** [comment]
> The situation here is that we agreed not to include user identifiers such as phone-numbers in the path, so we need to send them as request body. In that case, we can't use a GET and we would need a POST. If we use POST with the resource-name it would be confusing because that would mean that we're creating a resource which is not the case here. So that's why we need to use a verb to avoid the confussion. Using a POST when you are not creating wouldn't be REST-ful either. You're right that if we used a GET we should have used nouns to make it resource-centric but in the current situation I think this is the best way to define it.

Agree, using post, but we're not doing a creating stuff, so confusion appears.

**Comment by monamok** [comment]
The decision about GET vs POST was made to have a similar solution to other CAMARA APIs and as some operators prefered not to send sensitive information as GET path/query parameters.
Currently there is an open issue discussing some of these points in commonalities:

https://github.com/camaraproject/WorkingGroups/issues/212

**Comment by gregory1g** [comment]
> Currently there is an open issue discussing some of these points in commonalities:
> 
> [camaraproject/WorkingGroups#212](https://github.com/camaraproject/WorkingGroups/issues/212)

Great, this approach should be clarified on the cross - API level! Thank you for sharing the link,

**Comment by DT-DawidWroblewski** [comment]
clarified
